container/kvm/initialisation.go

```diff
@@ -56,3 +84,117 @@ func ensureDependencies() error {
 
 	return nil
 }
+
+// createPool creates the libvirt storage pool directory. runCmd and chownFunc
+// are here for testing. runCmd so we can check the right shell out calls are
+// made, and chownFunc because we cannot chown unless we are root.
+func createPool(pathfinder func(string) (string, error), runCmd runFunc, chownFunc func(string) error) error {
+	poolDir, err := guestPath(pathfinder)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	if err = definePool(poolDir, runCmd, chownFunc); err != nil {
+		return errors.Trace(err)
+	}
+	if err = buildPool(runCmd); err != nil {
+		return errors.Trace(err)
+	}
+
+	if err = startPool(runCmd); err != nil {
+		return errors.Trace(err)
+	}
+	if err = autostartPool(runCmd); err != nil {
+		return errors.Trace(err)
+	}
+
+	// We have to set ownership of the guest pool directory after running virsh
+	// commands above, because it appears that the libvirt-bin version that
+	// ships with trusty sets the ownership of the pool directory to the user
+	// running the commands -- root in our case. Which causes container
+	// initialization to fail as we couldn't write volumes to the pool. We
+	// write them as libvirt-qemu:kvm so that libvirt -- which runs as that
+	// user -- can read them to boot the domains.
+	if err = chownFunc(poolDir); err != nil {
+		return errors.Trace(err)
+	}
+
+	return nil
+}
+
+// definePool creates the required directories and changes ownershipt of the
+// guest directory so that libvirt-qemu can read, write, and execute its
+// guest volumes.
+func definePool(dir string, runCmd runFunc, chownFunc func(string) error) error {
+	// Permissions gleaned from https://goo.gl/SZIw14
+	// The command itself would change the permissions to match anyhow.
+	err := os.MkdirAll(dir, 0755)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	output, err := runCmd(
+		"virsh",
+		"pool-define-as",
+		poolName,
+		"dir",
+		"-", "-", "-", "-",
+		dir)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	logger.Debugf("pool-define-as ouput %s", output)
+
+	return nil
+}
+
+// chownToLibvirt changes ownership of the provided directory to
+// libvirt-qemu:kvm.
+func chownToLibvirt(dir string) error {
```

Comment:
I mean, I'm ok with this not being tested, but in general it would be better to mock out the thing we can't call and pass the fake one in - it reduces the amount of code that only runs in CI.